### DISCUSSION TITLE
Error: The 'login' command requires an interactive shell

### DISCUSSION THREAD

**Discussion by irvile** [planetscale/discussion#44] [Q&A] [answered]
Hi, 

There is a way to login without interactive shell?
i need run a script to execute pscales commands.

steps to reproduce:
1. yum localinstall pscale_0.39.0_linux_amd64.rpm
2. pscale auth login

any help are welcome.

**Comment by Anders-PlanetScale** [comment, marked as answer]
Howdy @irvile,

You can use [service tokens](https://docs.planetscale.com/reference/planetscale-cli#service-tokens-in-organizations) to avoid having to run auth login interactively. You will need create these from pscale CLI so you will have to interactively authenticate initially.

The following doc utilizes the service_token
https://planetscale.com/docs/tutorials/connect-any-application

You can pass the --service-token-name <token-name> --service-token <token> at the end of most pscale commands to auth without doing auth login.

There is also some more discussion around headless pscale here;

https://github.com/planetscale/beta/discussions/11

**Reply by irvile** [reply]
Thanks for your answer!

cant wait for next steps. Im really enjoying pscale.

**Reply by default-student** [reply]
link 404s.
new link:
https://planetscale.com/docs/tutorials/connect-any-application
